kernel/kernel-api/src/main/java/io/delta/kernel/commit/CatalogCommitterUtils.java

```diff
@@ -70,7 +61,7 @@ public static Map<String, String> extractProtocolProperties(Protocol protocol) {
     if (protocol.supportsReaderFeatures() || protocol.supportsWriterFeatures()) {
       for (String featureName : protocol.getReaderAndWriterFeatures()) {
         properties.put(
-            TableFeatures.SET_TABLE_FEATURE_SUPPORTED_PREFIX + featureName,
+            TableFeatures.getTableFeature(featureName).getTablePropertyElementKey(),
```

Comment:
This is where you _do_ want to do dynamic lookup, because we are iterating through a list of feature names!